File: lintcode/524_left-pad/left-pad.py
# coding:utf-8
'''
@Copyright:LintCode
@Author:   chenweikang
@Problem:  http://www.lintcode.com/problem/left-pad
@Language: Python
@Datetime: 16-04-25 20:36
'''

class StringUtils:
    # @param {string} originalStr the string we want to append to
    # @param {int} size the target length of the string
    # @param {string} padChar the character to pad to the left side of the string
    # @return {string} a string
    @classmethod
    def leftPad(self, originalStr, size, padChar=' '):
        return padChar * (size - len(originalStr)) + originalStr
        

File: lintcode/524_left-pad/left-pad.cpp
/*
@Copyright:LintCode
@Author:   chenweikang
@Problem:  http://www.lintcode.com/problem/left-pad
@Language: C++
@Datetime: 16-04-25 20:34
*/

class StringUtils {
public:
    /**
     * @param originalStr the string we want to append to
     * @param size the target length of the string
     * @param padChar the character to pad to the left side of the string
     * @return a string
     */
    static string leftPad(string& originalStr, int size, char padChar=' ') {
        if (size > originalStr.size())
            return string(size-originalStr.size(), padChar) + originalStr;
        return originalStr;
    }
};